src/org/zaproxy/zap/extension/fuzz/payloads/generator/RegexPayloadGenerator.java

```diff
@@ -71,26 +78,198 @@ public long getNumberOfPayloads() {
 
     @Override
     public ResettableAutoCloseableIterator<StringPayload> iterator() {
-        List<String> payloads = new RegExStringGenerator().regexExpansion(regex, maxPayloadLength, maxPayloads);
-        List<StringPayload> tempPayloads = new ArrayList<>(payloads.size());
-        for (String payload : payloads) {
-            tempPayloads.add(new DefaultStringPayload(payload));
-        }
-        return new PayloadCollectionIterator<>(tempPayloads);
+        return new RegexIterator(generator, maxPayloads);
     }
 
     @Override
     public RegexPayloadGenerator copy() {
-        return new RegexPayloadGenerator(regex, maxPayloads, maxPayloadLength, numberOfPayloads);
+        return this;
+    }
+
+    /**
+     * Tells whether or not the given {@code regex} is syntactically valid for the payload generator.
```

Comment:
I would change this to "Tells whether or not the given {@code regex} has the proper syntax for the payload generator."